apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -1,72 +1,152 @@
+      variablesExample,
+      commandControlValues || {}
     );
 
     return {
-      issues: validatedContent.issues,
       result: {
         preview: executeOutput.outputs as any,
         type: stepData.type as unknown as ChannelTypeEnum,
       },
-      previewPayloadExample: validatedContent.finalPayload,
-    };
+      previewPayloadExample: variablesExample,
+    } as GeneratePreviewResponseDto;
   }
 
   @Instrument()
-  private async getValidatedContent(dto: GeneratePreviewRequestDto, stepData: StepDataDto, user: UserSessionData) {
-    if (!stepData.controls?.dataSchema) {
-      throw new StepMissingControlsException(stepData.stepId, stepData);
+  private buildVariablesExample(
+    workflow: WorkflowInternalResponseDto,
+    stepData: StepDataDto,
+    commandVariablesExample?: PreviewPayload,
+    commandControlValues?: Record<string, unknown>
+  ) {
+    const variablesExample = this.generateVariablesExample(stepData, commandControlValues);
+
+    if (workflow.origin === WorkflowOriginEnum.EXTERNAL) {
+      variablesExample.payload = this.generateSamplePayload(workflow.payloadSchema);
     }
 
-    return await this.prepareAndValidateContentUsecase.execute({
-      stepType: stepData.type,
-      controlValues: dto.controlValues || stepData.controls.values,
-      controlDataSchema: stepData.controls.dataSchema,
-      variableSchema: stepData.variables,
-      previewPayloadFromDto: dto.previewPayload,
-      user,
+    return _.merge(variablesExample, commandVariablesExample as Record<string, unknown>);
+  }
+
+  @Instrument()
+  private generateVariablesExample(stepData: StepDataDto, commandControlValues?: Record<string, unknown>) {
+    const controlValues = Object.values(commandControlValues || stepData.controls.values).join('');
+    const variablesExample = pathsToObject(extractTemplateVars(controlValues), {
+      valuePrefix: '{{',
+      valueSuffix: '}}',
     });
+
+    return variablesExample;
+  }
+
+  @Instrument()
+  private async findWorkflow(command: GeneratePreviewCommand) {
+    return await this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        identifierOrInternalId: command.identifierOrInternalId,
+        environmentId: command.user.environmentId,
+        organizationId: command.user.organizationId,
+        userId: command.user._id,
+      })
+    );
+  }
+
+  /**
+   * Generates a payload based solely on the schema.
+   * Supports nested schemas and applies defaults where defined.
+   * @param JSONSchemaDto - Defining the structure. example:
+   *  {
+   *    firstName: { type: 'string', default: 'John' },
+   *    lastName: { type: 'string' }
+   *  }
+   * @returns - Generated payload. example: { firstName: 'John', lastName: '{{payload.lastName}}' }
+   */
+  @Instrument()
+  private generateSamplePayload(schema: JSONSchemaDto, path = 'payload', depth = 0): Record<string, unknown> {
+    const MAX_DEPTH = 10;
+    if (depth >= MAX_DEPTH) {
+      return {};
```

Comment:
```suggestion
      throw new Error('Schema has surpassed the maximum allowed depth (10). Please specify a more shallow payload schema.');
```

We need a BadRequestException exception for that.